src/lightspeed_rag_content/asciidoc/asciidoctor_converter.py

```diff
@@ -0,0 +1,185 @@
+Please read: https://docs.asciidoctor.org/asciidoctor/latest/extensions/
+You can also investigate the default text converter 'asciidoc_text_converter.rb'
+stored in the asciidoc package.
+"""
+
+import logging
+import shutil
+import subprocess
+from importlib import resources
+from pathlib import Path
+
+import yaml
+
+LOG: logging.Logger = logging.getLogger(__name__)
+
+RUBY_ASCIIDOC_DIR: Path = Path(resources.files(__package__)).joinpath("ruby_asciidoc")
+
+
+class AsciidoctorConverter:
+    """Convert AsciiDoc formatted documents to different formats.
+
+    The class requires asciidoctor to be installed. By default, all files are
+    converted to text format using a custom asciidoctor compatible extension
+    that is written in Ruby.
+    """
+
+    def __init__(
+        self,
+        target_format: str = "text",
+        attributes_file: Path | None = None,
+        converter_file: Path | None = None,
+    ):
+        """Initialize AsciidoctorConverter.
+
+        Args:
+            target_format:
+                A format to which input files should be converted. These formats
+                are currently supported: text, html5, xhtml5, manpage.
+            attributes_file: A path pointing to an attributes file.
+            converter_file: An asciidoctor compatible extension.
+
+        Raises:
+            FileNotFoundError:
+                When asciidoctor executable or converter_file cannot be found.
+
+            yaml.YAMLError:
+                When attributes_file is not valid YAML file.
+        """
+        self.target_format = target_format
+        self.attribute_list = self._get_attribute_list(attributes_file)
+
+        if not converter_file:
+            self.converter_file = self._get_converter_file(target_format)
+        else:
+            self.converter_file = converter_file
+
+        self.asciidoctor_cmd = self._get_asciidoctor_path()
+
+    @staticmethod
+    def _get_converter_file(target_format: str) -> Path | None:
+        """Return converter file if target_format requires one."""
+        asciidoctor_supported_formats = ["html5", "xhtml5", "manpage"]
+        if target_format in asciidoctor_supported_formats:
+            return None
+
+        converter_files = {
+            "text": "asciidoc_text_converter.rb",
+        }
+
+        if not (converter_file := converter_files.get(target_format, None)):
+            raise FileNotFoundError(
+                f"There is no extension available for target format: {target_format}"
+            )
+
+        return RUBY_ASCIIDOC_DIR.joinpath(converter_file)
+
+    @staticmethod
+    def _get_asciidoctor_path() -> str:
+        """Check whether asciidoctor and ruby are installed."""
+        asciidoctor_path = shutil.which("asciidoctor")
+        if not asciidoctor_path:
+            raise FileNotFoundError("asciidoctor executable not found")
+
+        LOG.info(f"Using asciidoctor with {asciidoctor_path} path")
```

Comment:
A nit and perhaps not worth fixing, but usually I don't like interpolating the log strings directly, I would rather leave it for the logging module to do as it will skip formatting the message if the log level is not enabled.

e.g

LOG.info("Using asciidoctor with %s path", asciidoctor_path)

Reply:
Thank you! It's useful. I did not even think about this.

Fixed.